system-test/system_test_redis_integ.sh

```
@@ -0,0 +1,1230 @@
+    echo -e "${GREEN}  ✓ Signal handling working correctly${NC}"
+    echo -e "${GREEN}  ✓ Signal metrics captured successfully${NC}"
+    echo -e "${GREEN}  ✓ Redis key expiration verified${NC}"
+    echo -e "${GREEN}========================================${NC}"
+}
+
+# Run main function with timeout in CI mode
+# Only execute main if this script is being run directly (not sourced)
+if [ "${BASH_SOURCE[0]}" = "${0}" ]; then
+    if [ "$CI_MODE" = true ]; then
+        if command -v timeout >/dev/null 2>&1; then
+            timeout "$TIMEOUT" bash -c "source '$0'; main '$@'" || {
```

Shellcheck (0.10.0)

[error] Argument mixes string and array. Use * or separate argument.

(SC2145)